modules/ROOT/pages/batch-processing-concept.adoc

```diff
@@ -192,32 +205,62 @@ During this phase, you can optionally configure the runtime to create a report o
-* Reference the batch job result object elsewhere in the Mule application to capture and use batch metadata, such as the number of records which failed to process in a particular batch job instance.
+The fields in the report object (a `BatchJobResult`) are accessible as keys when using DataWeave selectors, such as `payload.failedRecords` to return the number of failed records in the instance.
 
-If you leave the On Complete phase empty and do not reference the batch job result object elsewhere in your application, the batch job simply completes, whether failed or successful.
+If you leave the On Complete phase empty and do not reference the batch job result object from elsewhere in your application, the batch job simply completes, whether failed or successful. However, after Mule completes execution of an entire batch job instance, the logs provide some processing information, for example:
 
-[TIP]
---
-As a good practice, you should configure some mechanism for reporting on failed or successful records to facilitate further action where required.
---
+----
+Finished execution for instance 'e84b5da0-fd5a-11ec-84a5-147ddaaf4f97'
+of job 'w-batch-take6Batch_Job'.
+Total Records processed: 1000. Successful records: 1000. Failed Records: 0
+----
 
+[[variable_propagation]]
 === Variable Propagation
 
-Every processed record of the Batch Job Instance starts with the same variables and values present before the execution of the block. Every record has its own set of variables, so new variables or modifications of already-existing variables during the processing of a given record will not be visible while processing another record. For each record, those variables (and modifications) are propagated through the different Batch Steps. For example, if record R1 sets a variable `varName: "hello"`, record R2 sets `varName: "world"`, and record R3 does not set this variable, then in the next step, R1 will see the value `"hello"`, R2 the value `"world"` and R3 will not see any value for that variable.
+Each record in a batch job instance inherits event variables from the input to the Batch Job scope. You can access and change the value of these variables in the Batch Step and Batch Aggregator scopes, which run during the Process phase, and you can create new variables within these scopes. For each record within this phase, the variables (and modifications to them) are propagated through the different Batch Step and Aggregator scopes within a Batch Job scope. For example, if record R1 sets a variable `varName: "hello"`, record R2 sets `varName: "world"`, and record R3 does not set this variable, then in the next step, R1 has the value `"hello"`, R2 the value `"world"`, and R3 has no value for that variable.
+//TODO: does the variable _exist_ (without a value set) in R3 though? Or does it not exist?
```

Comment:
//TODO: does the variable _exist_ (without a value set) in R3 though? Or does it not exist?

Reply:
OR DOES IT DEPEND ON HOW THE VARIABLE WAS CREATED?